docs/guides/stretch.mdx

```diff
@@ -0,0 +1,162 @@
+
+## Example
+
+A common use case of `stretch` is to apply dynamical decoupling to an idling qubit while another qubit is undergoing conditional operations.
+
+For example, we can use `stretch` to apply an XX dynamical decoupling sequence to qubit 1, for the duration of the conditional block applied to qubit 0, as illustrated by the following diagram:
+
+![Image illustrating the following circuit](/docs/images/guides/stretch/stretch.avif)
+
+The corresponding circuit would look like the following. Note that a pair of barriers is needed to define the boundaries of this relative timing.
+
+```python
+from qiskit.circuit import QuantumCircuit, QuantumRegister, ClassicalRegister
+
+qubits = QuantumRegister(2)
+clbits = ClassicalRegister(2)
+circuit = QuantumCircuit(qubits, clbits)
+(q0, q1) = qubits
+(c0, c1) = clbits
+
+# Add barriers to define the boundaries
+circuit.barrier()
+circuit.h(q0)
+circuit.measure(q0, c0)
+with circuit.if_test((c0, 1)) as else_:
+    circuit.h(q0)
+with else_:
+    circuit.x(q0)
+
+# Apply an XX DD sequence with stretch on qubit 1
+s = circuit.add_stretch("s")
+circuit.delay(s, q1)
+circuit.x(q1)
+circuit.delay(expr.mul(s, 2), q1)
+circuit.x(q1)
+circuit.delay(s, q1)
+circuit.barrier()
+```
+
+Here is another example of using `stretch` to ensure a sequence of gates between two barriers are left-aligned, whatever their actual durations are:
+```python
+from qiskit import QuantumCircuit
+from numpy import pi
+
+qc = QuantumCircuit(5)
+qc.barrier()
+qc.cx(0, 1)
+qc.u(pi/4, 0, pi/2, 2)
+qc.cx(3, 4)
+
+a = qc.add_stretch("a")
+b = qc.add_stretch("b")
+c = qc.add_stretch("c")
+
+# Use the stretches as Delay duration.
+qc.delay(a, [0, 1])
+qc.delay(b, 2)
+qc.delay(c, [3, 4])
+qc.barrier()
+```
+
+## Limitations
+
+The following constraints apply when using `stretch`:
+
+- At most one stretch variable per _qubit set_ between barriers (implicit and explicit). A qubit set is one or more qubits; these sets must be mutually exclusive.
+    Valid:
+
+        ```python
+        stretch a;
+        stretch b;
+        delay[a] $0, $1;
+        delay[b] $2
+        ```
+
+    Invalid:
+
+        ```python
+        stretch a;
+        stretch b;
+        delay[a] $0, $1;
+        delay[b] $0;
+        ```
+
+- A stretch variable cannot be used between multiple sets of barriers.
+        Valid:
+
+        ```python
+        barrier $0, $1;
+        delay[a] $0;
+        barrier $3;
+        delay[a] $0;
+        barrier $0, $1;
+        ```
+
+    Invalid:
+
+        ```python
+        barrier $0, $1;
+        delay[a] $0;
+        barrier $0, $1;
+        delay[a] $0;
+        barrier $0, $1;
+         ```
+
+- Stretch expressions are limited to those of the form `A*stretch + B` where `A` and `B` are floating point or integer constants.
+        Valid:
+
+        ```python
+        delay[2 * a + 3.0] $0;
```

Comment:
@jakelishman do you know how to do this in Qiskit? I tried `expr.add(expr.mul(a, 2), 3.0)` but that didn't work. 

Actually I think it's the `add`, because  a simple `expr.add(a, 2)` also doesn't work.

Reply:
ah yes this makes sense. Since I can do `delay(3)` with an implicit unit of `dt`, and a stretch also has a unit of `dt` (once resolved), I thought `expr.add(a, 2)` would work.

Reply:
The Qiskit `Delay` instruction predates all the classical-expression handling, and it (and tbh all the rest of Qiskit) never really handled dimensioned quantities safely - lots of the code frequently forgot to check the unit, which was stored completely separately to the amount, even though the unit is an absolutely critical component of the quantity itself - the amount is meaningless without a unit.  The new `qiskit.circuit.Duration` represents a duration literal safely, including its literal (like OQ3 also always requires a suffix denoting the unit of duration literals).  The old handling of `Delay` will eventually be phased out.